### ISSUE TITLE
[PATCH] \includegraphics redefinition is broken in sphinx.sty

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#213]
I noticed some problems when trying to include centered figures using raw latex for the figure but reST images (or images included using the graphviz extension), see http://groups.google.com/group/sphinx-dev/browse_thread/thread/2beb7aef4739f739/db445196c9e607bf for details.

I tracked down the problem and found out that is the sphinx.sty redefinition of the tex command \includegraphics:

```
&#37; Redefine includgraphics for avoiding images larger than the screen size
&#37; If the size is not specified.
\let\py@Oldincludegraphics\includegraphics

\newbox\image@box&#37;
\newdimen\image@width&#37;
\renewcommand\includegraphics[2][\[False](https://bitbucket.org/empty)]{&#37;
  \ifx#1\[False](https://bitbucket.org/empty)&#37;
    \setbox\image@box=\hbox{\py@Oldincludegraphics{#2}}&#37;
    \image@width\wd\image@box&#37;
    \ifdim \image@width>\linewidth&#37;
      \setbox\image@box=\hbox{\py@Oldincludegraphics[width=\linewidth]{#2}}&#37;
    \fi&#37;
    \box\image@box&#37;
  \else&#37;  
    \py@Oldincludegraphics[#1]{#2}&#37;
  \fi&#37;    
}       
```

If I comment that code in sphinx.sty, I get the expected results (the image centered in the figure). I don't know how useful is that redefinition, maybe it's doing more harm than good (maybe not, I barely know LaTeX =).

Attached is a pure-latex minimal example to reproduce the bug/solution, just run pdflatex test.tex (you need sphinx.sty and manual.cls) and see the results before and after commenting the offending code in sphinx.sty.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/213
- Originally reported by: Leandro Lucarella
- Originally created at: 2009-07-04T22:28:38.002

**Comment by shimizukawa** [comment]
_From Leandro Lucarella on 2009-07-04 20:43:34+00:00_

Attached are a patch and a bundle that fixes the command redefinition. Thanks to Maximiliano Curia for the LaTeX fix!

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2009-07-09 09:57:08+00:00_

Applied the fix in <<cset b809740f6613cf2810e8f935749b2beb456f1b95>>. Thanks!

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2013-03-31 13:57:10+00:00_

Removing version: 0.6.2 (automated comment)
